linux_os/guide/system/software/updating/ensure_redhat_gpgkey_installed/oval/shared.xml

```diff
@@ -24,16 +26,24 @@
           test_ref="test_package_gpgkey-{{{ aux_pkg_version }}}-{{{ aux_pkg_release }}}_installed" />
         </criteria>
       </criteria>
-      <criteria comment="CentOS Vendor Keys" operator="OR">
+      {{%- if product == "rhel6" %}}
+      <criteria comment="CentOS Vendor Keys" operator="AND">
         <criteria comment="CentOS Installed" operator="OR">
           <extend_definition comment="CentOS6 installed" definition_ref="installed_OS_is_centos6" />
+        </criteria>
+        <criterion comment="package gpg-pubkey-c105b9de-4e0fd3a3 is installed"
+        test_ref="test_package_gpgkey-c105b9de-4e0fd3a3_installed" />
+      </criteria>
+      {{%- endif %}}
+      {{%- if product == "rhel7" or product == "rhv4" %}}
+      <criteria comment="CentOS Vendor Keys" operator="AND">
+        <criteria comment="CentOS Installed" operator="OR">
```

Comment:
Same as for CentOS7, this OR criteria "CentOS Installed" is not needed.

Reply:
Removed